User data

Guptacos opened the pull request:
Add getter methods to the User class for various user information

Guptacos committed:
Add user data getters to User class

Guptacos committed:
Add User.image() to get profile picture

Guptacos committed:
Add private data getters to User class

Add email(), subscription(), and country()

Guptacos commented:
The images branch must be merged before this one

rmannava commented on spotifython/user.py:
```diff
@@ -19,6 +22,11 @@ class User:
+    Unsupported:
+
+        * external_urls
+        * followers: unsupported by Spotify, see https://developer.spotify.com/documentation/web-api/reference/object-model/#followers-object
```
google style guide says to put the url on its own line of its too long, might not need the pylint disable

Guptacos replied:
It's too long even on its own line; this way at least it renders nicely with sphinx

rmannava commented on spotifython/user.py:
```diff
@@ -40,6 +48,71 @@ def __init__(self, session, info):
+        # Updates _raw with new values. One liner : for each key in union of
+        # keys in self._raw and response_json, takes value for key from
+        # response_json if present, else takes value for key from self._raw.
+        # TODO: this is weird notation, make a utility function for it.
+        # Especially useful since it is an action necessary for many classes.
+        self._raw = {**self._raw, **response_json}
```
would self._raw.update(response_json) do the same thing?

spenceryu replied:
I use this magic method across a lot of my code too. If we change this, it should be a separate PR

Guptacos replied:
Yea it does see the groupchat

rmannava commented on spotifython/user.py:
```diff
@@ -40,6 +48,71 @@ def __init__(self, session, info):
+
+        # We know other is a 'User' object, so this protected access is okay
+        #pylint: disable=protected-access
+        self._raw = {**self._raw, **other._raw}
```
same as above

Guptacos committed:
Change {**x, **y} to x.update(y) in User class

Does the same thing but is more readable. The only difference is that
update() updates x in place (instead of returning a new object) which is
fine in our case.
